core/translate/expr.rs

```diff
@@ -689,100 +752,41 @@ pub fn translate_expr(
                 #[cfg(feature = "json")]
                 Func::Json(j) => match j {
                     JsonFunc::Json => {
-                        let args = if let Some(args) = args {
-                            if args.len() != 1 {
-                                crate::bail_parse_error!(
-                                    "{} function with not exactly 1 argument",
-                                    j.to_string()
-                                );
-                            }
-                            args
-                        } else {
-                            crate::bail_parse_error!(
-                                "{} function with no arguments",
-                                j.to_string()
-                            );
-                        };
-                        let regs = program.alloc_register();
-                        translate_expr(program, referenced_tables, &args[0], regs, resolver)?;
-                        program.emit_insn(Insn::Function {
-                            constant_mask: 0,
-                            start_reg: regs,
-                            dest: target_register,
-                            func: func_ctx,
-                        });
-                        Ok(target_register)
-                    }
-                    JsonFunc::JsonArray => {
-                        let start_reg = translate_variable_sized_function_parameter_list(
+                        let args = expect_arguments_exact!(args, 1, j);
+
+                        translate_function(
                             program,
                             args,
                             referenced_tables,
                             resolver,
-                        )?;
-
-                        program.emit_insn(Insn::Function {
-                            constant_mask: 0,
-                            start_reg,
-                            dest: target_register,
-                            func: func_ctx,
-                        });
-                        Ok(target_register)
+                            target_register,
+                            func_ctx,
+                        )
                     }
-                    JsonFunc::JsonExtract => {
-                        let start_reg = translate_variable_sized_function_parameter_list(
+                    JsonFunc::JsonArray | JsonFunc::JsonExtract => translate_function(
```

Comment:
Cleaned it up a little. Do you want me to move it to a separate PR?

Reply:
nah let's just blast